### ISSUE TITLE
Make Unicode-awareness of convertToRaw optional

### ISSUE THREAD

**Issue by robbertbrak** [facebook/draft-js#1351]
I'm using `convertToRaw` to convert the Draft `contentState` to Markdown in Javascript. However, it turns out that this method is "smart" about Unicode when describing the offsets and lengths of inline style ranges, which is _not_ what I want.

For example, when you type an emoji in the supplementary plane such as 💩 in the rich text sample and make it bold, `convertToRaw` will report that there is a `BOLD` inline style range of length 1.

This is ok when the raw `contentState` is postprocessed in some other programming language, but I find this very confusing for Javascript, in which the 💩 emoji is usually counted as two characters.

Would it be ok to add an option to `convertToRaw` to control how Unicode is treated? If so, I'd be happy to submit a pull request for it.

**Comment by flarnie** [comment]
Hi @robbertbrak - Thanks for flagging this!

**tldr:** Yes this is a valid problem, but we would prefer not to add any more API surface area to `convert<To/From>Raw`, so this would likely work best as a module outside of Draft.

**More conversational version:**
There does seem to be a common use case here - just now I searched and found several libraries for converting Draft EditorState to-and-from markdown, and we also do this internally at Facebook. I would be curious how they have handled this, or if it's also a problem for them.

Ideally there is a solution amongst those implementations, which perhaps could be of use to you. If they all also suffer from this issue, then I think there is room in the ecosystem for a fork of `convertToRaw` which supports non-unicode-aware parsing of the text.

I am [trying to avoid having a boolean config option](https://martinfowler.com/bliki/FlagArgument.html) that changes how the `convertToRaw` method works, and also trying to keep the Draft API small. Even internally at Facebook, I see many different formats which people want to convert to and from when saving Draft state - many teams build their own layer around `convertToRaw` and `convertFromRaw` for this reason.
